sdk/bpf/scripts/install.sh

```diff
@@ -9,77 +9,78 @@ else
 fi
 
 download() {
-  set -e
   declare url="$1/$2/$3"
-  declare version=$2
   declare filename=$3
-  declare dirname=$4
-  declare progress=$5
-  declare cache_directory=~/.cache/"$version"
-  declare cache_dirname=$cache_directory/${dirname//:\//_}
-
-  link() {
-    set -e
-    ln -sf "$cache_dirname" "$dirname"
-    ln -sf "$cache_directory/$dirname-$version.md" "$dirname-$version.md"
-  }
-
-  if [[ -r $cache_dirname ]]; then
-    link
-    return 0
-  fi
-
-  declare args=(
+  declare wget_args=(
     "$url" -O "$filename"
-    "--progress=dot:$progress"
+    "--progress=dot:giga"
     "--retry-connrefused"
     "--read-timeout=30"
   )
+  declare curl_args=(
+    -L "$url" -o "$filename"
+  )
+  wget_or_curl=$(
+    if hash wget 2>/dev/null; then
+      echo "wget ${wget_args[*]}"
+    elif hash curl 2>/dev/null; then
+      echo "curl ${curl_args[*]}"
+    fi
+  )
+  if [[ -z $wget_or_curl ]]; then
+    echo "Error: Neither curl nor wget were found" >&2
+    return 1
+  fi
   set -x
-  mkdir -p "$cache_dirname"
-  pushd "$cache_dirname"
-  if wget "${args[@]}"; then
-    tar --strip-components 1 -jxf "$filename"
+  if $wget_or_curl; then
+    tar --strip-components 1 -jxf "$filename" || return 1
+    { set +x; } 2>/dev/null
     rm -rf "$filename"
-    echo "$url" >"../$dirname-$version.md"
-    popd
-    link
     return 0
   fi
-  popd
-  rm -rf "$cache_dirname"
   return 1
 }
 
 clone() {
-  set -e
   declare url=$1
   declare version=$2
-  declare dirname=$3
-  declare cache_directory=~/.cache/"$version"
-  declare cache_dirname=$cache_directory/${dirname//:\//_}
 
-  link() {
-    set -e
-    ln -sf "$cache_dirname" "$dirname"
-    ln -sf "$cache_directory/$dirname-$version.md" "$dirname-$version.md"
-  }
+  rm -rf temp
+  declare cmd="git clone --recursive --depth 1 --single-branch --branch $version $url temp"
+  set -x
+  if $cmd; then
+    { set +x; } 2>/dev/null
+    shopt -s dotglob nullglob
```

Comment:
I'd either set `shopt` at the top for the entire script, or run in a subshell:
```
(
  shopt -s dotglob nullglob
  mv temp/* .
)
```
As it stands, `shopt` is not unset so it will be in effect even after the function returns